jdtls.ext/com.microsoft.java.maven.plugin/src/main/java/com/microsoft/java/maven/AddDependencyHandler.java

```diff
@@ -0,0 +1,266 @@
+
+        public AddDependencyParams(String fullClassName, String artifactInfo, 
+                String uri, int line, int character, int length) {
+            this.fullClassName = fullClassName;
+            this.artifactInfo = artifactInfo;
+            this.uri = uri;
+            this.line = line;
+            this.character = character;
+            this.length = length;
+        }
+    }
+
+    private static WorkspaceEdit importEdit(AddDependencyParams params, 
+            ICompilationUnit unit, CompilationUnit astRoot) {
+        // import the new class
+        final CompilationUnitChange cuChange = new CompilationUnitChange("", unit);
+        final ImportRewrite importRewrite = CodeStyleConfiguration.createImportRewrite(astRoot, true);
+        importRewrite.addImport(params.fullClassName);
+        try {
+            cuChange.setEdit(importRewrite.rewriteImports(null));
+            return ChangeUtil.convertToWorkspaceEdit(cuChange);
+        } catch (CoreException e) {
+            return new WorkspaceEdit();
+        }
+    }
+
+    private static WorkspaceEdit replaceEdit(AddDependencyParams params, ICompilationUnit unit) {
+        // replace old class name with new one
+        try {
+            final Map<String, List<TextEdit>> textEdits = new HashMap<>();
+            textEdits.put(unit.getCorrespondingResource().getLocationURI().toString(),
+                    Arrays.asList(new TextEdit(new Range(new Position(params.line, params.character), 
+                    new Position(params.line, params.character + params.length)), 
+                    params.fullClassName.substring(params.fullClassName.lastIndexOf('.') + 1))));
+            return new WorkspaceEdit(textEdits);
+        } catch (JavaModelException e) {
+            return new WorkspaceEdit();
+        }
+    }
+
+    private static PosInfo getPosInfo(String pomPath, String targetDependency) {
+        PosInfo posInfo = null;
+        final SAXParserFactory factory = SAXParserFactory.newInstance();
+        try {
+            final SAXParser parser = factory.newSAXParser();
+            final XMLReader reader = parser.getXMLReader();
+            final GetPosHandler posHandler = new GetPosHandler();
+            posHandler.setTargetDependency(targetDependency);
+            reader.setContentHandler(posHandler);
+            reader.parse(pomPath);
+            posInfo = posHandler.getPosInfo();
+            return posInfo;
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            return null;
+        }
+    }
+
+    private static WorkspaceEdit pomEdit(AddDependencyParams params, ICompilationUnit unit) {
+        IPath path = null;
+        try {
+            path = unit.getJavaProject().getCorrespondingResource().getLocation().append("pom.xml");
+            final String info[] = params.artifactInfo.replaceAll(" ", "").split(":");
+            final PosInfo posInfo = getPosInfo(path.toString(), info[0] + ":" + info[1]);
```

Comment:
you are implicitly asserting info.length >= 2

Reply:
maybe I should check if info.length >= 2. return empty workspaceedit if it is not satisfied?